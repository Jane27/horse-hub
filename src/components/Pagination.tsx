import React from "react";
import styled from "styled-components";

type TPagination = {
  total: number;
  currPage: number;
  setPageContent: (index: number) => void;
};

const Container = styled.div`
  display: flex;
  flex-direction: row;
`;

const Button = styled.button<{ active?: boolean }>`
  color: ${(props) => (props.active ? "#fff" : "#000")};
  background-color: ${(props) => (props.active ? "blue" : "#fff")};
`;

const Pagination: React.FC<TPagination> = ({
  total: itemsNo,
  currPage,
  setPageContent,
}) => {
  const totalPages = Math.ceil(itemsNo / 10);
  const renderSerial = (total: number) => {
    const buttonGroup: any[] = [];

    for (let i = 0; i < total; i++) {
      buttonGroup.push(
        <Button active={i === currPage} onClick={() => setPageContent(i)}>
          {i + 1}
        </Button>
      );
    }
    return buttonGroup;
  };

  return (
    <Container>
      <button
        disabled={currPage === 0}
        onClick={() => setPageContent(currPage - 1)}
      >
        Previous
      </button>
      {renderSerial(totalPages)}
      <button
        disabled={currPage === totalPages - 1}
        onClick={() => setPageContent(currPage + 1)}
      >
        Next
      </button>
    </Container>
  );
};

export default Pagination;
