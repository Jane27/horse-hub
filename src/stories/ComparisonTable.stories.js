import React from 'react';
import { withKnobs, object } from '@storybook/addon-knobs/react';
import { ComparisonTable } from '../components';

export default {
  title: 'ComparisonTable',
  decorators: [withKnobs],
  component: ComparisonTable,
};

const fields = [
  {
    label: "Name",
    path: "name",
  },
  {
    label: "Favourite Food",
    path: "profile.favouriteFood",
  },
  {
    label: "Height",
    path: "profile.physical.height",
  },
  {
    label: "Weight",
    path: "profile.physical.weight",
  },
];

const options = [
  {
    name: "Red boy",
    profile: {
      favouriteFood: "apple",
      physical: {
        height: 300,
        weight: 60
      }
    }
  },
  {
    name: "Purple girl",
    profile: {
      favouriteFood: "orange",
      physical: {
        height: 70,
        weight: 250
      }
    }
  },
  {
    name: "Green girl",
    profile: {
      favouriteFood: "orange",
      physical: {
        height: 70,
        weight: 250
      }
    }
  }
]




export const CompareTItems = () => {
  const horse = object ("horse", options)
  const compareFields = object ("compareFields", fields)

return <ComparisonTable fields={compareFields} items={horse} />};


