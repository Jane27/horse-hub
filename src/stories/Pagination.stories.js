import React from 'react';
import { withKnobs, number } from '@storybook/addon-knobs/react';
import { Pagination } from '../components';


export default {
  title: 'Pagination',
  decorators: [withKnobs],
  component: Pagination,
};


export const simple = () =>{
  const total = number("total", 48);
  const currentPage = number("currentPage", 0);  
  const pageSize = number("pageSize", 10);
return <Pagination total={total} initialPage={currentPage} pageSize={pageSize} >Hello Button</Pagination>;

} 

