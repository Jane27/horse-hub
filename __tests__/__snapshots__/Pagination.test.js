import { shallow } from 'enzyme';
import { simple } from '../../src/stories/Pagination.stories'

describe('pagination', () => {
  it('Should have previous , next button, the first number button is active', () => {
    let output = shallow(simple());
    expect(output).toMatchSnapshot();
    expect(output.find('button').first().text()).toEqual('Previous');
    expect(output.find('NumberButton')).toHaveLength(5);
    expect(output.find('NumberButton').first().props().active).toEqual(true);
    expect(output.find('NumberButton').last().props().active).toEqual(false);
    expect(output.find('button').last().text()).toEqual('Next');
  });
});

