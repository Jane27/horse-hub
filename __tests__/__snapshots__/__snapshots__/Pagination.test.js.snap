// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination Should have previous , next button, the first number button is active 1`] = `
<styled.div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Previous
  </button>
  <NumberButton
    active={true}
    key="0"
    onClick={[Function]}
  >
    1
  </NumberButton>
  <NumberButton
    active={false}
    key="1"
    onClick={[Function]}
  >
    2
  </NumberButton>
  <NumberButton
    active={false}
    key="2"
    onClick={[Function]}
  >
    3
  </NumberButton>
  <NumberButton
    active={false}
    key="3"
    onClick={[Function]}
  >
    4
  </NumberButton>
  <NumberButton
    active={false}
    key="4"
    onClick={[Function]}
  >
    5
  </NumberButton>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next
  </button>
</styled.div>
`;
